components/eamxx/src/share/grid/remap/coarsening_remapper.cpp

```diff
@@ -291,6 +304,33 @@ do_bind_field (const int ifield, const field_type& src, const field_type& tgt)
 
 void CoarseningRemapper::do_registration_ends ()
 {
+  // Before finishing the registration we need to also register the set of masks, if needed.
+  if (m_track_mask) {
+    for (int idx=0; idx<m_mask_fields_src.size(); ++idx) {
+      auto f   = m_mask_fields_src[idx];
+      auto fid = f.get_header().get_identifier();
+      register_field_from_src(fid);
+      m_mask_map_tgt.emplace(idx,m_num_registered_fields-1);
+    }
+    // Update the number of fields.
+    m_num_fields = m_num_registered_fields;
```

Comment:
My understanding was that the number of registered fields is calculated first, before `do_registration_ends()` is called.  Since here we are actually adding fields we have to update the number.  See: https://github.com/E3SM-Project/scream/blob/master/components/eamxx/src/share/grid/remap/abstract_remapper.cpp#L97

Maybe the more correct thing would be to change `do_registration_ends()` in the abstract_remapper class?

Reply:
Ah right, yes.